### PR TITLE
Support Job for Openshift, Knative and Kubernetes

### PR DESCRIPTION
**Pull request by Sgitario**
More information in the feature request: https://github.com/dekorateio/dekorate/issues/856
Fix https://github.com/dekorateio/dekorate/issues/856

### CODE REVIEW

File: docs/documentation/kubernetes.md [hunk 1]
```diff
@@ -292,6 +292,34 @@ Currently, the supported annotations for specifying volumes are:
...
+  }
+}
+```    
+
+Or via configuration properties at the file `application.properties`:
+
+    dekorate.kubernetes.jobs[0].name=say-hello
+    dekorate.kubernetes.jobs[0].containers[0].image=docker.io/user/hello
+
+Currently, the supported annotations for adding Kubernetes Jobs are:
```
Comment (aureamunoz) [inline docs/documentation/kubernetes.md:317]:
```suggestion
Currently, the supported annotations for adding jobs are:
```

File: docs/documentation/kubernetes.md [hunk 2]
```diff
@@ -292,6 +292,34 @@ Currently, the supported annotations for specifying volumes are:
...
+
+- @KubernetesApplication
+- @OpenShiftApplication
+- @KnativeApplication
```
Comment (aureamunoz) [inline docs/documentation/kubernetes.md:321]: It should be nice to add this paragraph also to the Knative and OpenShift `docs/documentation/md` files, wdyt?

**Review by aureamunoz — CHANGES_REQUESTED** [review]

**Review by iocanel — CHANGES_REQUESTED** [review]
I am not sure if we should get into the business of defining arbitrary jobs via dekorate.
When I first heard about Job support in Kubernetes, I thought that the goal is to expose the application in development as a Job, rather than being able to define side jobs.

Let's meet and discuss.

**Comment by Sgitario** [comment]
> I am not sure if we should get into the business of defining arbitrary jobs via dekorate. When I first heard about Job support in Kubernetes, I thought that the goal is to expose the application in development as a Job, rather than being able to define side jobs.
> 
> Let's meet and discuss.

As discussed, for Jobs, we're going to discuss two use cases:
- The generation of side Jobs. This is implemented as part of this PR. 
- Allow setting the kind output which will be implemented in https://github.com/dekorateio/dekorate/issues/886.

**Comment by Sgitario** [comment]
@iocanel I've already updated this PR with the support of volumes.

**Review by iocanel — APPROVED** [review]
LGTM

**Pull request merged**